lib/sensu-plugin/utils.rb

```diff
@@ -33,6 +41,82 @@ def read_event(file)
         exit 0
       end
 
+      ##
+      #  Helper method to convert Sensu 2.0 event into Sensu 1.4 event
+      #    This is here to help keep Sensu Plugin community handlers working
+      #    until they natively support 2.0
+      #    Takes 2.0 event json object as argument
+      #    Returns event with 1.4 mapping included
+      #
+      #    Note:
+      #      The 1.4 mapping overwrites some attributes so the resulting event cannot
+      #      be used in a 2.0 workflow. The top level boolean attribute "v2_event_mapped_into_v1"
+      #      will be set to true as a hint to indicate this is a mapped event object.
+      #
+      ##
+      def map_v2_event_into_v1(orig_event = nil)
+        orig_event ||= @event
+
+        # return orig_event if already mapped
+        return orig_event if orig_event['v2_event_mapped_into_v1']
+
+        # Deep copy of orig_event
+        event = Marshal.load(Marshal.dump(orig_event))
+
+        # Trigger mapping code iff enity exists and client does not
+        client_missing = event['client'].nil? || event['client'].empty?
+        if event.key?('entity') && client_missing
+          ##
+          # create the client hash from the entity hash
+          ##
+          event['client'] = event['entity']
+
+          ##
+          # Fill in missing client attributes
+          ##
+          event['client']['name']        ||= event['entity']['id']
+          event['client']['subscribers'] ||= event['entity']['subscriptions']
+
+          ##
+          # Fill in renamed check attributes expected in 1.4 event
+          #   subscribers, source
+          ##
+          event['check']['subscribers'] ||= event['check']['subscriptions']
+          event['check']['source'] ||= event['check']['proxy_entity_id']
+
+          ##
+          # Mimic 1.4 event action based on 2.0 event state
+          #  action used in logs and fluentd plugins handlers
+          ##
+          state = event['check']['state'] || 'unknown::2.0_event'
+          event['action'] ||= 'flapping' if state.casecmp('flapping').zero?
+          event['action'] ||= 'resolve' if state.casecmp('passing').zero?
+          event['action'] ||= 'create' if state.casecmp('failing').zero?
+          event['action'] ||= state
+
+          ##
+          # Mimic 1.4 event history based on 2.0 event history
+          #  Note: This overwrites the same history attribute
+          #    2.x history is an array of hashes, each hash includes status
+          #    1.x history is an array of statuses
+          ##
+          if event['check']['history']
+            legacy_history = []
+            event['check']['history'].each do |h|
+              legacy_history << h['status'].to_s || '3'
```

Comment:
hmm if we are not sure what the history is I feel like we should use `nil` (though that might cause issues?) as not knowing how to translate is not quite the same as the check itself returning an `unknown` state.

Reply:
Also is history really stored as the string when really they are integers?

Reply:
For the 1.4 events I have access to, the are stored as strings.
there is at least one mutator in the sensu-plugins collection that makes use of the history, and in fact makes a comment that the history is saved as an array strings.

sensu-plugins-pagerduty/bin/mutator-pagerduty-priority-override.rb

Returning nil instead of '3' may cause weird issues  nil.to_i gets translated to 0 which is the 'ok' status. I think if we don't know how to translate the status in the history its better to use the 3 aka 'unknown' to make sure we flag it as a problem that needs to be investigated.

Or it may be better to throw an error here instead of providing a fallback if the status is unmappable to a string.